src/components/KExternalLink/KExternalLink.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+  <a
+    v-if="href"
+    class="k-external-link"
+    :href="href"
+    rel="noopener"
+    target="_blank"
+  >
+    <slot />
+    <KIcon
+      v-if="!hideIcon"
+      color="var(--blue-500)"
+      icon="externalLink"
+      size="12"
+    />
+  </a>
+</template>
+
+<script setup lang="ts">
+defineProps({
+  href: {
+    type: String,
+    required: true,
+    validator: function(value:string) {
+      const urlPattern = new RegExp('^(https?:\\/\\/)?' + // protocol
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+    '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+    '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+    '(\\#[-a-z\\d_]*)?$', 'i') // fragment locator
+
+      return !!urlPattern.test(value)
+    },
```

Comment:
suggestion: Instead of using this RegEx (unsure where it came from originally) we could instead try constructing a URL with a try/catch within the validator. Something like this:
```suggestion
    validator: (url: string): boolean => {
      try {
        const linkUrl = new URL(url)
        
        // No error was thrown, so return true (we could check some of the URL pieces here, but I'm not sure it's necessary)        
        return true
      } catch (err) {
        // Invalid URL
        return false
      }
    },
```